### ISSUE TITLE
False positive: explicit_self with Swift property wrappers

### ISSUE THREAD

**Issue by rebello95** [realm/SwiftLint#3075]
### Describe the bug

When using Swift property wrappers, auto-correcting the `explicit_self` rule yields adding a `self.` in the wrong location.

Before:

`print($property)`

After autocorrect:

`print($self.property)`

Expected:

`print(self.$property)`

Additionally, it identifies a false positive in a valid situation:

`print(self.$property)`

And attempts to autocorrect to:

`print(self.$self.property)`

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint analyze --compiler-log-path /Users/mrebello/Downloads/output.txt --config /Users/mrebello/Downloads/SwiftlintTest/.swiftlint.yml --path /Users/mrebello/Downloads/SwiftlintTest/SwiftlintTest/AppDelegate.swift --autocorrect
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.38.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
analyzer_rules:
  - explicit_self
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? 11.3.1
* Do you have a sample that shows the issue?

```swift
@UIApplicationMain
final class AppDelegate: UIResponder, UIApplicationDelegate {
    var window: UIWindow?

    func application(_ application: UIApplication,
                     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?)
        -> Bool
    {
        SomeClass().doSomething()
        return true
    }
}

@propertyWrapper
struct CustomWrapper {
    private var value: String

    init(wrappedValue: String) {
        self.value = wrappedValue
    }

    var wrappedValue: String {
        get { return self.value + "1" }
        set { self.value = newValue }
    }

    var projectedValue: String {
        return self.wrappedValue
    }
}

final class SomeClass {
    @CustomWrapper
    private var property = "foo"

    func doSomething() {
        // Incorrectly autocorrects to print($self.property)
        print($property)
        // Incorrectly autocorrects to print(self.$self.property)
        print(self.$property)
    }
}
```

**Comment by keith** [comment]
Still relevant
